File: rules/php/tests/test_deserialization_tp.php
<?php
// True positive: unserialize on user input
class DataController extends Controller
{
    public function import(Request $request)
    {
        // ruleid: ez-laravel-insecure-deserialization
        $obj = unserialize($request->input('data'));
        return response()->json($obj);
    }
}

File: rules/java/tests/test_deserialization_tp.java
// True positive: ObjectInputStream from request
import java.io.ObjectInputStream;
import javax.servlet.http.HttpServletRequest;

public class DataImporter {
    public Object importData(HttpServletRequest req) throws Exception {
        // ruleid: ez-spring-insecure-deserialization
        ObjectInputStream ois = new ObjectInputStream(req.getInputStream());
        return ois.readObject();
    }
}
